src/slurm_plugin/clustermgtd.py

```diff
@@ -956,17 +965,30 @@ def _handle_protected_mode_process(self, active_nodes, partitions_name_map):
                 "command to re-enable the fleet."
             )
 
-    @staticmethod
-    def _handle_nodes_failing_health_check(nodes_failing_health_check, health_check_type):
+    def _handle_nodes_failing_health_check(self, nodes_failing_health_check: List[SlurmNode], health_check_type: str):
         # Place unhealthy node into drain, this operation is idempotent
         if nodes_failing_health_check:
-            nodes_name_failing_health_check = {node.name for node in nodes_failing_health_check}
-            log.warning(
-                "Setting nodes failing health check type %s to DRAIN: %s",
-                health_check_type,
-                nodes_name_failing_health_check,
-            )
-            set_nodes_drain(nodes_name_failing_health_check, reason=f"Node failing {health_check_type}")
+            nodes_name_failing_health_check = set()
+            nodes_name_recently_rebooted = set()
+            for node in nodes_failing_health_check:
+                if not node.is_reboot_issued() and time_is_up(
+                    node.slurmdstarttime, self._current_time, self._config.health_check_timeout_after_slurmdstarttime
+                ):
+                    nodes_name_failing_health_check.add(node.name)
+                else:
+                    nodes_name_recently_rebooted.add(node.name)
+            if len(nodes_name_failing_health_check) > 0:
+                log.warning(
+                    "Setting nodes failing health check type %s to DRAIN: %s",
+                    health_check_type,
+                    nodes_name_failing_health_check,
+                )
+                set_nodes_drain(nodes_name_failing_health_check, reason=f"Node failing {health_check_type}")
+            if len(nodes_name_recently_rebooted) > 0:
+                log.info(
+                    "These nodes were recently restarted and they were not set to DRAIN: %s",
```

Comment:
What about changing `restarted` to rebooted?

Reply:
Minor discussion here: strictly speaking, this condition may occur even if someone manually restarts `slurmd` on a compute node, without completely rebooting the compute node. I believe I had originally written something like `On these nodes the slurmd daemon was recently restarted...`, and that's why I kept the restarted word here.

Also, this condition may also occur the first time ever a compute node is booted, so _rebooted_ or _restarted_ may be the wrong words. 

Should we write something like: `The slurmd service was recently started on these nodes and therefore they were not set to DRAIN`? Do we like it or is it too convoluted?